lambda_functions/vetext_incoming_forwarder_lambda/vetext_incoming_forwarder_lambda.py

```diff
@@ -96,28 +95,49 @@ def process_body_from_sqs_invocation(event):
         # event["body"] is a base 64 encoded string
         # parse_qsl converts url-encoded strings to array of tuple objects
         # event_body takes the array of tuples and creates a dictionary
-        try:
-            event_body_decoded = parse_qsl(b64decode(record["body"]).decode('utf-8'))
-            event_body = dict(event_body_decoded)
-            event_bodies.append(event_body)
-        except:
-            push_to_sqs(record["body"])
+        try: 
+            event_body = record.get("body", "")
 
+            if (event_body == ""):
+                logger.info("event_body from sqs record was not present")
+                logger.debug(record)
+                continue
+
+            logger.debug("Processing record body from SQS: %s", event_body)
+            event_body = json.loads(event_body)
+            logger.info("Successfully converted record body from sqs to json")
+            event_bodies.append(event_body)
+        except Exception as e:
+            logger.exception(e)        
+            logger.info("Failed to load event from sqs")
+            push_to_sqs(event_body)
+    
     return event_bodies
 
 def process_body_from_alb_invocation(event):
-    event_bodies = []
-
     # event is a json document with a body attribute that contains
     #   the payload of the twilio webhook
     # event["body"] is a base 64 encoded string
     # parse_qsl converts url-encoded strings to array of tuple objects
     # event_body takes the array of tuples and creates a dictionary
-    event_body_decoded = parse_qsl(b64decode(event["body"]).decode('utf-8'))
-    event_bodies.append(dict(event_body_decoded))
+    event_body_encoded = event.get("body", "")
 
-    return event_bodies
-    
+    if (event_body_encoded == ""):
+        logger.info("event_body from alb record was not present")
+        logger.debug(event)        
+
+    event_body_decoded = parse_qsl(b64decode(event_body_encoded).decode('utf-8'))
+    logger.info(f"Decoded event_body {event_body_decoded}")
+
+    event_body = dict(event_body_decoded)
+    logger.info("Converted body to dictionary")
+
+    if 'AddOns' in event_body:        
+        logger.info(f"AddOns present in event_body: {event_body['AddOns']}")
+        del event_body['AddOns']
```

Comment:
```suggestion
        event_body.pop('AddOns', None)
```

Reply:
Del will result in a key error if it's not there. Use [pop](https://docs.python.org/3/library/stdtypes.html#dict.pop).